src/sentry/snuba/metrics/fields/base.py

```diff
@@ -299,16 +296,16 @@ def validate_can_orderby(self) -> None:
     def run_post_query_function(
         self,
         data: SnubaDataType,
-        metrics_query: MetricsQuery,
         metric_mri: str,
         alias: str,
         idx: Optional[int] = None,
+        params: Optional[MetricOperationParams] = None,
     ) -> SnubaDataType:
         raise NotImplementedError
 
     @abstractmethod
     def generate_filter_snql_conditions(
-        self, org_id: int, metrics_query: MetricsQuery, use_case_id: UseCaseKey
+        self, org_id: int, use_case_id: UseCaseKey, params: Optional[MetricOperationParams] = None
```

Comment:
```suggestion
        self, org_id: int, use_case_id: UseCaseKey, params: MetricOperationParams = None
```
isn't `MetricOperationParams` already `Optional`?

So we have `Optional[Optional[Mapping[...]]]]` instead of `Optional[Mapping[str, Union[str, int, float]]]`?

Reply:
oops, yeah you are right! I got confused in the middle as well. 
Will fix